File: test-login.py
#!/usr/bin/env python3
import requests
import json

def test_login():
    url = "http://localhost:8000/api/auth/login"
    data = {
        "username": "officer@agri.gov.in",
        "password": "officer123", 
        "role": "officer"
    }
    
    try:
        response = requests.post(url, json=data, headers={"Content-Type": "application/json"})
        print(f"Status Code: {response.status_code}")
        print(f"Response Headers: {dict(response.headers)}")
        print(f"Response Body: {response.text}")
        
        if response.status_code == 200:
            result = response.json()
            print(f"Success! Token: {result.get('access_token', 'N/A')}")
            print(f"User: {result.get('user', {})}")
        else:
            print("Login failed!")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    test_login()

File: test-login.js
import fetch from 'node-fetch';

const testLogin = async () => {
  try {
    const response = await fetch('http://localhost:8000/api/auth/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        username: 'officer@agri.gov.in',
        password: 'officer123',
        role: 'officer'
      })
    });

    const data = await response.json();
    console.log('Login response:', data);
    console.log('Status:', response.status);
  } catch (error) {
    console.error('Login error:', error);
  }
};

testLogin();
